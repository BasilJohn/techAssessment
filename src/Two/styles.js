import {StyleSheet, Dimensions} from 'react-native';
import * as colorConstants from '../constants/colorConstants';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  buttonContainer: {
    width: '90%',
    height: Dimensions.get('window').height * 0.1,
    backgroundColor: colorConstants.navigationButtonColor,
    alignSelf: 'center',
    marginBottom: 10,
  },
  button: {
    height: '100%',
    width: '100%',
    justifyContent: 'center',
  },
  textStyle: {
    textAlign: 'center',
  },
  welcomeViewContainer: {
    width: '100%',
    height: Dimensions.get('window').height * 0.05,
    justifyContent: 'center',
    alignItems: 'flex-end',
    paddingRight: 20,
  },
});
