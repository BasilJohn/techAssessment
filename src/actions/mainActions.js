/**
 * @format
 * @flow strict-local
 */

export const setInputValue = (inputValue: string) => ({
  type: 'SET_INPUT_VALUE',
  payload: inputValue,
});

export const setEmulatorInformation = (isEmulator: string) => ({
  type: 'SET_EMULATOR_INFO',
  payload: isEmulator,
});
