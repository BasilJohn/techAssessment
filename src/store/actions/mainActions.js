/**
 * @format
 * @flow strict-local
 */

const setInputValue = (inputValue: string) => ({
  type: 'SET_INPUT_VALUE',
  payload: inputValue,
});

const setEmulatorInformation = (isEmulator: string) => ({
  type: 'SET_EMULATOR_INFO',
  payload: isEmulator,
});

export { setInputValue, setEmulatorInformation };
