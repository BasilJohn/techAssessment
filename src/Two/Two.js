/**
 * @format
 * @flow strict-local
 */
import React, {Component} from 'react';
import {useSelector} from 'react-redux';

import {View, Text, TouchableOpacity} from 'react-native';
import {styles} from './styles';
import {NavigationScreenProps} from '@react-navigation/native';

type TwoProps = NavigationScreenProps & {};


const Two = (props: TwoProps) => {

  const navigateToScreen = (screenName: string) => {
    props.navigation.navigate(screenName);
  };

  
  const value = useSelector(state => state.main.inputValue);
  return (
    <View style={styles.container}>
      <View style={styles.welcomeViewContainer}>
        <Text>Input Data :{value}</Text>
      </View>
      <View style={styles.buttonContainer}>
        <TouchableOpacity
          testID={"btnScreenOne"}
          onPress={() => navigateToScreen('One')}
          style={styles.button}>
          <Text style={styles.textStyle}>Go to screen one</Text>
        </TouchableOpacity>
      </View>
      <View style={styles.buttonContainer}>
        <TouchableOpacity
          onPress={() => navigateToScreen('Three')}
          style={styles.button}>
          <Text style={styles.textStyle}>Go to screen three</Text>
        </TouchableOpacity>
      </View>
    </View>
  );
};

export default Two;
