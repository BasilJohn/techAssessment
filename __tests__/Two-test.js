/**
 * @format
 */

import {TouchableOpacity} from 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Two from '../src/Two/Two';
import mainReducer from '../src/reducers/reducer';
import configureMockStore from 'redux-mock-store';
import {setInputValue} from '../src/actions/mainActions';
import {NativeModules} from 'react-native';
import {Provider} from 'react-redux';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import * as redux from 'react-redux';

const mockStore = configureMockStore([]);
const store = mockStore({
  main: {
    inputValue: '',
    isEmulator: 'false',
  },
});

const comp = renderer.create(
  <Provider store={store}>
    <Two />
  </Provider>,
);
test('Renders Component Two Correctly', () => {
  const tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

