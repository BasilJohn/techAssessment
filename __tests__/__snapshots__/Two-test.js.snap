// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Component Two Correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "height": 66.7,
        "justifyContent": "center",
        "paddingRight": 20,
        "width": "100%",
      }
    }
  >
    <Text>
      Input Data :
      
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "grey",
        "height": 133.4,
        "marginBottom": 10,
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
      testID="btnScreenOne"
    >
      <Text
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Go to screen one
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "grey",
        "height": 133.4,
        "marginBottom": 10,
        "width": "90%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "justifyContent": "center",
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Go to screen three
      </Text>
    </View>
  </View>
</View>
`;
